README.md

```diff
@@ -1,3 +1,142 @@
+This repository contains an up-to-date fork of
+[Wasmtime](https://docs.wasmtime.dev/), a standalone WebAssembly engine. This
+fork adds support for the [WasmFX](https://wasmfx.dev/) proposal for stack
+switching. 
+
+The [stack switching repository](https://github.com/WebAssembly/stack-switching) contains a 
+[high-level summary of the proposal](https://github.com/WebAssembly/stack-switching/blob/main/proposals/continuations/Explainer.md),
+[a more formal description](https://github.com/WebAssembly/stack-switching/blob/main/proposals/continuations/Overview.md),
+and [examples](https://github.com/WebAssembly/stack-switching/tree/main/proposals/continuations/examples).
+
+## Building 
+
+The build steps are equivalent to the [standard steps for building Wasmtime from
+source](https://docs.wasmtime.dev/contributing-building.html), but using this
+repository instead. There is no need to build or install the original version of
+Wasmtime to use this fork.
+
+Concretely, the steps are as follows:
+
+1. Make sure that you have a Rust toolchain installed, for example using
+   [rustup](https://www.rust-lang.org/tools/install).
+2. Check out this repository:
+``` sh
+git clone https://github.com/wasmfx/wasmfxtime.git
+cd wasmfxtime
+git submodule update --init
+```
+3. Build:
+``` sh
+cargo build
+```
+
+As a result, a debug build of the `wasmtime` executable will be created at
+`target/debug/wasmtime`.
+
+To create a release build instead, run `cargo build --release`, which will
+create `target/release/wasmtime`.
+
+
+## Running programs
+
+A WebAssembly module `my_module.wat` (or `my_module.wasm`) is executed using the
+`wasmtime` executable [in the usual way](https://docs.wasmtime.dev/cli.html). To
+run programs containing WasmFX instructions, enable the necessary features as
+follows:
+
+``` sh
+wasmtime -W=exceptions,function-references,typed-continuations my_module.wat
+```
+
+To run an arbitrary function exported as `foo` by the module run 
+
+``` sh
+wasmtime -W=exceptions,function-references,typed-continuations --invoke=foo my_module.wat
+```
+
+## Example program
+
+The following module implements a generator and consumer using stack switching.
```

Comment:
I think we should give an entirely self-contained program without any indirection, e.g. "see examples/generator.wat". The reader should be able to copy the example verbatim and run it. I suggest we use an example such as https://dhil.net/research/talks/generator-waw24.wast

Reply:
The example is self-contained in the sense that you can copy-paste and run it, you just won't get the printed output. The example program you link to relies on the `spectest` module to do the the printing, which is only available when running wasmtime in `wast` mode for running tests. The instructions here are explicitly intended so that you can execute the module in the standard way.

Reply:
I am not happy with the current example. Maybe we should do a wast example for the output and then go into how to compile and run arbitrary programs. From a usability perspective I think it is important that you can copy paste the first example and observe a concrete output.

Reply:
I think that we should provide an example that can actually be executed in the same way that you would run an ordinary program. Having to discuss a second way of running programs (namely, in wast mode) seems worse to me than putting the full version of the example behind a link.
Let's see what @slindley thinks once he's back.